engine/lib/phases/phase_state_passing_loop.ml

```diff
@@ -0,0 +1,378 @@
+(* TODO: handle Exn report *)
```

Comment:
Mmh, this is basically a copy-paste of `phase_local_mutation` with a few differences, right?
We need to either make `phase_local_mutation` parametric or split it in two.

I recall we discussed that phase at some point.
If I'm not mistaking, you want to allow local mutation, but at the boundaries of functions and loops? i.e. you want to rewrite mutations of outer variables within a loop?

Currently, I think this `phase_state_passing_loop` is still rewriting all local mutation into shadowing bindings though! Or I'm mistaking?

Reply:
Yes, I did copy paste this from `phases_local_mutation`. I want to keep mutable variables (to some extent), the only changes I did was keeping the mutability, as I do not see why it was removed by this phase, however I could refactor it into two phases instead, and just keep the first for the SSProve backend.

Reply:
I might want to allow more mutability, as SSProve can actually reason about it. However, for now I just want to keep the mutable assignments throughout the translations, this is still translated to "shadowing" mutable let binding, but this is an assignment to a memory location by name, so it has the same effect.